### DISCUSSION TITLE
hanging when requesting image with `image/webp` accept header

### DISCUSSION THREAD

**Discussion by tomgrossman** [vercel/next.js#48242] [Help]
### Summary

This is a strange one, no idea what is going on here and how to even debug it, bear with me.

In the last days-week a part of our nextjs app pages started hanging (not loading completely).
We saw, that a request to get the background image of that page does not return at all - stuck.
Taking the request, as is, from the browser's network logs, and calling the same url manually with the same headers - the requests just hangs manually as well.

**HOWEVER 1** - When changing the automatic `accept` header that `next` automatically adds from `'Accept: image/webp'` to a different type like `'Accept: image/jpeg'` or just removing it, making the request returns successfully.

**HOWEVER 2** - When recreating the k8s pod of that app (same code, same image, just creating a new one) sometimes solves the issue, also with `'Accept: image/webp'`. Sometimes not (!!) but sometimes it helps. Weird, right?

So far we couldn't reproduce the issue locally, just in our k8s envs (all of them).

This is the component we are using that gets stuck:

```typescript
import { Box } from '@chakra-ui/react';
import Image from 'next/image';
import { PropsWithChildren } from 'react';
import DashboardImage from 'public/img/dashboard-bg.jpg';
import MspDashboardImage from 'public/img/MSP-dashboard-blur.jpg';

export type BlurredAppBackgroundProps = PropsWithChildren & {
  mspMode?: boolean;
};

const BlurredAppBackground = ({ children, mspMode }: BlurredAppBackgroundProps) => (
  <Box m={-6}>
    <Image alt="" style={{ width: '100%' }} priority src={mspMode ? MspDashboardImage : DashboardImage} />
    {children}
  </Box>
);

export default BlurredAppBackground;
```

this command gets stuck:
```bash
curl 'http://127.0.0.1:2222/_next/image?url=/_next/static/media/dashboard-bg.e8325f26.jpg&w=1920&q=75' \                    
  -H 'Accept: image/webp'
```
but this works:
```bash
curl 'http://127.0.0.1:222/_next/image?url=/_next/static/media/dashboard-bg.e8325f26.jpg&w=1920&q=75'
```

The `2222` is the port forwarded on the pod
After restarting the pod, sometimes both commands work, sometimes another restart is required.

There are no logs, errors or anything in the app, we have no idea how to debug it and to understand what is going on.
We are using the latest versions of `next, charka, sharp, react...`.

Please help us investigate it!!!

### Additional information

```js
npx next info (from the pod's container):

    Operating System:
      Platform: linux
      Arch: x64
      Version: #1 SMP Fri Dec 30 10:10:30 UTC 2022
    Binaries:
      Node: 18.15.0
      npm: 9.5.0
      Yarn: 1.22.19
      pnpm: N/A
    Relevant packages:
      next: 13.3.0
      eslint-config-next: 13.3.0
      react: 18.2.0
      react-dom: 18.2.0


sharp@0.32.0
```

**Comment by tomgrossman** [comment]
Right now we added `unoptimized` until someone figure it out 🤷‍♂️

**Comment by acschm1d** [comment]
This is still an issue in 14.2.13. While

`curl -I https://example.com/_next/image/\?url\=%2Fassets%2Fsomeasset%402x.png\&w\=256\&q\=75 -H 'Accept: text/html,application/xhtml+xml,application/xml;q=0.9,image/avif,image/webp,image/apng,*;q=0.8;'`

works,

 `curl -I https://example.com/_next/image/\?url\=%2Fassets%2Fsomeasset%402x.png\&w\=256\&q\=75 -H 'Accept: text/html,application/xhtml+xml,application/xml;q=0.9,image/avif,image/webp,image/apng,*/*;q=0.8;'`

does not.

Note the `*/*` which seems to cause the crash.

Doing just a request with

 `curl -I https://example.com/_next/image/\?url\=%2Fassets%2Fsomeasset%402x.png\&w\=256\&q\=75 -H 'Accept: */*;'`

will hence also crash, while

 `curl -I https://example.com/_next/image/\?url\=%2Fassets%2Fsomeasset%402x.png\&w\=256\&q\=75 -H 'Accept: *;'`

won't.

The library behind handling this is hapi.
